packages/react/src/components/Swap/Swap.tsx

```diff
@@ -33,7 +54,51 @@ export const Swap = ({ tokenBondingKey }: { tokenBondingKey: PublicKey }) => {
     tradingMints,
     onTradingMintsChange: setTradingMints,
     swap: (args) =>
-      execute(args).then(({ targetAmount }) => {
+      execute({
+        async extraInstructions({isFirst, amount, isBuy}: IExtraInstructionArgs): Promise<InstructionResult<null>> {
```

Comment:
If swap driver takes an id, shouldn't these extra instrs be added in useSwapDriver?